airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java

```diff
@@ -55,29 +55,29 @@ public Optional<SnowflakeTableDefinition> findExistingTable(StreamId id) throws
   }
 
   @Override
-  public boolean isFinalTableEmpty(StreamId id) throws SQLException {
-    int rowCount = database.queryInt(
+  public boolean isFinalTableEmpty(final StreamId id) throws SQLException {
+    final int rowCount = database.queryInt(
         """
         SELECT row_count
         FROM information_schema.tables
         WHERE table_catalog = ?
           AND table_schema = ?
           AND table_name = ?
         """,
-        databaseName,
+        databaseName.toUpperCase(),
```

Comment:
Will this break v2 stuff? Since we're not always using uppercase? Or is the DB always uppercase regardless? Would an `ilike()` be safer?

Reply:
tl;dr I think toUpperCase is correct

I ran `create database "edgao_MIXED_CASE_test_db";` to create a database with mixed-case name, granted ownership to AIRBYTE_ROLE, then put that value into the config. `check` actually failed (I’d assumed we would fail at sync time) on non-v2-related code.

Then I ran `create database edgao_MIXED_CASE_test_db;` (i.e. no quotes, so implicitly uppercased) + granted ownership. `check` succeeded in this case.

so I'm concluding that destination-snowflake is implicitly upcasing the database name everywhere.

(technical notes: I also had to `create schema edgao_mised_case_test_db.edgao_test`, and there was some annoying role-switching between sysadmin and airbyte_role to get all the permissions right. But none of that has bearing on the name-upcasing behavior.)